### ISSUE TITLE
Use of uninitialized value $sdev in right bitshift

### ISSUE THREAD

**Issue by heapdavid** [liske/needrestart#296]
Hi, we run `needrestart -b` on a cron and occasionally see the following output to stderr on some of our busier servers. We're struggling to reproduce reliably as it's pretty rare and intermittent.
```
Use of uninitialized value $sdev in right bitshift (>>) at /usr/sbin/needrestart line 532, <HMAP> line 2.
Use of uninitialized value $sdev in right bitshift (>>) at /usr/sbin/needrestart line 532, <HMAP> line 2.
Use of uninitialized value $sdev in bitwise and (&) at /usr/sbin/needrestart line 532, <HMAP> line 2.
Use of uninitialized value $sdev in right bitshift (>>) at /usr/sbin/needrestart line 532, <HMAP> line 2.
Use of uninitialized value $sdev in right bitshift (>>) at /usr/sbin/needrestart line 532, <HMAP> line 2.
Use of uninitialized value $sdev in bitwise and (&) at /usr/sbin/needrestart line 532, <HMAP> line 2.
Use of uninitialized value $sdev in right bitshift (>>) at /usr/sbin/needrestart line 544, <HMAP> line 2.
Use of uninitialized value $sdev in right bitshift (>>) at /usr/sbin/needrestart line 544, <HMAP> line 2.
Use of uninitialized value $sinode in string eq at /usr/sbin/needrestart line 549, <HMAP> line 2
```
Have seen there was a similar issue a while ago but that looked to be fixed at the time.

We're running Debian 12 (Bookworm)
```
$ apt list needrestart
Listing... Done
needrestart/stable,now 3.6-4 all [installed]
```
Please let me know if you need any further information

Thanks

**Comment by liske** [comment]
Needrestart tries to compares the inode information from `/proc/$PID/maps` with the corresponding files in the filesystem. For some reasons the stat call seems to fail for some reasons: https://github.com/liske/needrestart/blob/c62b1529c5af3a78ad209ff942b289d4b4cdb45a/needrestart#L555

The path is the first available: at https://github.com/liske/needrestart/blob/c62b1529c5af3a78ad209ff942b289d4b4cdb45a/needrestart#L543-L544

Do you have any special filesystem setup? This might also happen if the process terminates while needrestart is investigating the procfs. :thinking:

**Comment by onyxmaster** [comment]
Encountering the same here. Since we use needrestart on cron when it happens we usually can't check what's happening. It happens rarely so I believe it's some kind of a concurrency issue (map file exists in the check that is above, but then it's removed on restart/unmap/etc.). I tried to fix this in #314 but please see my description there.

**Comment by liske** [comment]
This should be closed by #314, feel free to reopen if it still happens with the upcoming needrestart 3.8 release.
